### PR TITLE
CI: Add 4.2-stable.expected file for GDExtension compatibility checks

### PR DESCRIPTION
**Pull request by akien-mga**
Adds the editor hint for `--validate-extension-api` to match `--dump-extension-api`, otherwise we're missing the EditorInterface singleton.

### CODE REVIEW

**Review by YuriSizov — APPROVED** [review]

**Comment by AThousandShips** [comment]
Should the 4.1 file not be renamed like the 4.0 one was?

**Comment by akien-mga** [comment]
We have the following errors:
```
2023-12-04 16:18:27 URL:https://raw.githubusercontent.com/godotengine/godot-cpp/godot-4.2-stable/gdextension/extension_api.json [5501700/5501700] -> "/tmp/tmp.WdNiCBDP1l" [1]
Godot Engine v4.3.dev.mono.custom_build.156b07f14 - https://godotengine.org/
 
Error: Validate extension JSON: API was removed: singletons/EditorInterface

2023-12-04 16:18:27 URL:https://raw.githubusercontent.com/godotengine/godot-cpp/godot-4.0-stable/gdextension/extension_api.json [5175722/5175722] -> "/tmp/tmp.EoFQLdx9mx" [1]
Godot Engine v4.3.dev.mono.custom_build.156b07f14 - https://godotengine.org/
 
ERROR: New API lacks base array: enums
   at: compare_dict_array (core/extension/extension_api_dump.cpp:1350)
2023-12-04 16:18:28 URL:https://raw.githubusercontent.com/godotengine/godot-cpp/godot-4.1-stable/gdextension/extension_api.json [5318978/5318978] -> "/tmp/tmp.MBIcrUqT41" [1]
Godot Engine v4.3.dev.mono.custom_build.156b07f14 - https://godotengine.org/
 
ERROR: New API lacks base array: enums
   at: compare_dict_array (core/extension/extension_api_dump.cpp:1350)
```

The `New API lacks base array: enums` errors are also reproducible in the current master branch, and don't seem to mark the check as failed.

`API was removed: singletons/EditorInterface` is new from adding the `4.2-stable` file though. From a quick look I don't understand why it's raised. Both the `4.2-stable` and the API json generated by my local build have that entry, yet I get this error.

CC @RedworkDE

**Comment by AThousandShips** [comment]
Could it be related in some way to that the version tag is updated after that? The `.json` file in godot-cpp is from the 4.2 branch pre updating to `4.3-dev`, unsure if it is used in the comparison for this

**Comment by akien-mga** [comment]
The script downloads the json from the godot-cpp tag of the version in the filename (first one if it's a range), so it should be comparing 4.3-dev (this PR) with 4.2-stable.
https://github.com/godotengine/godot/blob/master/misc/scripts/validate_extension_api.sh#L58-L61

File: main/main.cpp [hunk 1]
```diff
@@ -2987,24 +2987,28 @@ bool Main::start() {
...
+			OS::get_singleton()->set_exit_code(EXIT_SUCCESS);
+			return false;
+		}
 
-	if (validate_extension_api) {
-		bool valid = GDExtensionAPIDump::validate_extension_json_file(validate_extension_api_file) == OK;
-		OS::get_singleton()->set_exit_code(valid ? EXIT_SUCCESS : EXIT_FAILURE);
-		return false;
+		if (validate_extension_api) {
+			OS::get_singleton()->set_editor_hint(true); // "extension_api.json" should always contains editor singletons.
```
Comment (akien-mga) [inline main/main.cpp:3007]: This fixes the issue that `singletons/EditorInterface` was missing in the API generated by `--validation-extension-api`, thus failing the comparison with the file generated by `--dump-extension-api`, which uses editor hint.

The rest is just a cosmetic change to keep GDExtension stuff visually grouped together.

File: main/main.cpp [hunk 2]
```diff
@@ -2987,24 +2987,28 @@ bool Main::start() {
...
+			OS::get_singleton()->set_exit_code(EXIT_SUCCESS);
+			return false;
+		}
 
-	if (validate_extension_api) {
-		bool valid = GDExtensionAPIDump::validate_extension_json_file(validate_extension_api_file) == OK;
-		OS::get_singleton()->set_exit_code(valid ? EXIT_SUCCESS : EXIT_FAILURE);
-		return false;
+		if (validate_extension_api) {
+			Engine::get_singleton()->set_editor_hint(true); // "extension_api.json" should always contains editor singletons.
```
Comment (akien-mga) [inline main/main.cpp:3007]: This fixes the issue that `singletons/EditorInterface` was missing in the API generated by `--validation-extension-api`, thus failing the comparison with the file generated by `--dump-extension-api`, which uses editor hint.

The rest is just a cosmetic change to keep GDExtension stuff visually grouped together.

**Pull request merged**